content/en/users/compute/orchestration/im/cli/_index.md

```diff
@@ -106,7 +109,42 @@ configure wn (
 deploy node 1
 ```
 
-Then we can call the `create` operation of the IM client tool:
+IM also supports TOSCA. For example this is an equivalent TOSCA document to
+deploy a single VM::
+
+```yaml
+tosca_definitions_version: tosca_simple_yaml_1_0
+
+imports:
+- indigo_custom_types: https://raw.githubusercontent.com/indigo-dc/tosca-types/master/custom_types.yaml
+
+topology_template:
+
+node_templates:
```

Comment:
Thanks @micafer !

Does `node_template` need to be indented below `topology_template`?

Reply:
When I copied and pasted the `yaml` and got:
```
$ im_client.py create tosca.yaml 
Secure connection with: https://appsgrycap.i3m.upv.es:31443/im/
ERROR creating the infrastructure: Error Creating Inf.: 'ToscaTemplate' object has no attribute 'nodetemplates'
```

When I execute it with the following `tosca.yaml`:
```yaml
tosca_definitions_version: tosca_simple_yaml_1_0
imports:
- indigo_custom_types: https://raw.githubusercontent.com/indigo-dc/tosca-types/master/custom_types.yaml
topology_template:
node_templates:
    simple_node:
      type: tosca.nodes.indigo.Compute
      capabilities:
        endpoint:
          properties:
            network_name: PUBLIC
        host:
          properties:
            num_cpus: 1
            mem_size: 1 GB
        os:
          properties:
            image: appdb://egi.ubuntu.20.04?vo.access.egi.eu
outputs:
  node_ip:
    value: { get_attribute: [ simple_node, public_address, 0 ] }
  node_creds:
    value: { get_attribute: [ simple_node, endpoint, credential, 0 ] }
```

I get:
```
$ im_client.py create tosca.yaml 
Secure connection with: https://appsgrycap.i3m.upv.es:31443/im/
ERROR creating the infrastructure: Error Creating Inf.: Error adding VMs: Error launching the VMs of type simple_node to cloud ID egi of type EGI. Error, no concrete system to deploy: simple_node in cloud: egi. Check if a correct image is being used
```

Sorry in advance for naive questions, I am just learning TOSCA now.

Reply:
> Does `node_template` need to be indented below `topology_template`?
You are right. I have fixed the indentation.

Reply:
Great!

I am still getting:
```
ERROR creating the infrastructure: Error Creating Inf.: Error adding VMs: Error launching the VMs of type simple_node to cloud ID egi of type EGI. Error, no concrete system to deploy: simple_node in cloud: egi. Check if a correct image is being used
```

Do you have a suggestion to solve this issue?

Reply:
You have detected a bug in the IM.
I will fix for the next release.
In the meantime you can change the image url to:
```yaml
image: appdb://SITE_NAME/egi.ubuntu.20.04?vo.access.egi.eu
```

Reply:
Oh, I see.

After updating the `tosca.yaml` file I get
```shell
$ im_client.py create tosca.yaml 
Secure connection with: https://appsgrycap.i3m.upv.es:31443/im/
ERROR creating the infrastructure: Error Creating Inf.: Project vo.access.egi.eu not found
```
Sorry for being picky, maybe I am doing something wrong now?

Reply:
are you part of the `vo.access.egi.eu` and the site supports it?

Reply:
Indeed, I am a member of `vo.access.egi.eu` and `GSI-LCG2` supports it. Actually, I have deployed VMs with IM Dashboard in (`vo.access.egi.eu`, `GSI-LCG2`) but I got the error above with `im-client`. Changing the site to `CESGA` in `im-client` worked! Maybe it has to do with the issues with TLS and IGTF certificates?

Reply:
Actually, I think the error above happens when the OpenStack project name differs from the VO name. I have replicated the issue with `IFCA-LCG2`.

Reply:
FYI: helper command is `fedcloud endpoint vos -a`

Reply:
Indeed it is bug related with the previous one.
I fixed it in the devel instance of the IM you can test it.
https://appsgrycap.i3m.upv.es:31443/im-dev/

Reply:
I get a `404` with https://appsgrycap.i3m.upv.es:31443/im-dev/
Please have a look and I will be happy to test.

Reply:
Oh, sorry, that's the REST API endpoint! Trying now!

Reply:
yep, working now!